src/wrapper.rs

```diff
@@ -77,8 +77,14 @@ where
         let mut e = self.e;
         let mut pattern = pattern.as_ref().to_vec();
         for &c in pattern.iter().rev() {
-            s = self.backend.lf_map2(c, s);
-            e = self.backend.lf_map2(c, e);
+            let s_mapped = self.backend.lf_map2(c, s);
+            let e_mapped = self.backend.lf_map2(c, e);
+            if let (Some(s_mapped), Some(e_mapped)) = (s_mapped, e_mapped) {
+                s = s_mapped;
+                e = e_mapped;
+            } else {
+                todo!("return a Result instead of panicking");
```

Comment:
We will run into this when `\0` is included in the given search pattern. To deal with this exception safely, we may want to return `Result` instead of panicking.

This will introduce breaking-changes to the API signature, so I'll work on it in another PR.